### PR TITLE
Add integration test for VM/VMX stack walkers for incomplete frame edge cases

### PR DESCRIPTION
**Pull request by Baraa-Hasheesh**
### Description
Adding a new integration test for VM & VMX stack walkers against certain edge cases

### Related issues
N/A

### Motivation and context
Increase coverage of tests in async profiler

### How has this been tested?
`make test`

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: test/native/libs/jninativestacks.c [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    while (i < 100000000) {
+        i++;
+        result += sqrt(i);
+        result += pow(i, sqrt(i));
+    }
+    return result;
+}
+
+__attribute__((unused)) JNIEXPORT double generateDeepStack(int count) {
```
Comment (apangin) [inline test/native/libs/jninativestacks.c:21]: Why `__attribute__((unused))`?
Compiler should never warn about _exported_ symbols in a shared library,

File: test/native/libs/jninativestacks.c [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    (void)frame;
+
+    if (count == 0) {
+        return doCpuTask();
+    } else {
+        return generateDeepStack(count - 1);
+    }
+}
+
+__attribute__((unused)) JNIEXPORT jdouble JNICALL Java_test_stackwalker_StackGenerator_generateLargeFrame(JNIEnv* env, jclass cls) {
```
Comment (apangin) [inline test/native/libs/jninativestacks.c:33]: Naming could be better. Can be simply `largeFrame`, since these functions do not _generate_ frames.

File: test/native/libs/jninativestacks.c [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+__attribute__((unused)) JNIEXPORT jdouble JNICALL Java_test_stackwalker_StackGenerator_generateDeepStack(JNIEnv* env, jclass cls) {
+    char frame[0x30000];
+    // Prevent from being optimized by compiler
+    (void)frame;
+    return generateDeepStack(6);
+}
+
+JNIEXPORT jdouble JNICALL Java_test_stackwalker_StackGenerator_generateCompleteStack(JNIEnv* env, jclass cls) {
```
Comment (apangin) [inline test/native/libs/jninativestacks.c:47]: In what sense it is "complete"? Perhaps you meant "leafFrame"?

File: test/one/profiler/test/TestProcess.java [hunk 4]
```diff
@@ -65,12 +65,14 @@ private static MethodHandle getPidHandle() {
     private final Process p;
     private final Map<String, File> tmpFiles = new HashMap<>();
     private final int timeout = 30;
+    private final Integer currentJvmVersion;
```
Comment (apangin) [inline test/one/profiler/test/TestProcess.java:68]: Do not use boxed numbers when primitives can be used.

File: test/test/stackwalker/StackGenerator.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+    public static native double generateDeepStack();
+
+    public static native double generateCompleteStack();
+
+    static {
+        System.loadLibrary("jninativestacks");
+    }
+
+    public static void main(String[] args) throws InterruptedException {
```
Comment (apangin) [inline test/test/stackwalker/StackGenerator.java:20]: Where is `InterruptedException` thrown?

File: test/test/stackwalker/StackwalkerTests.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    @Test(mainClass = StackGenerator.class, jvmArgs = "-Xss5m", args = "completeStack",
+            agentArgs = "start,event=cpu,cstack=vmx,file=%f.jfr")
+    public void normalStackVMX(TestProcess p) throws Exception {
+        p.waitForExit();
+        assert p.exitCode() == 0;
+        Output output = Output.convertJfrToCollapsed(p.getFilePath("%f"));
+        assert output.contains("^" +
+                FRAME + // Root can be different on different systems
+                OPTIONAL_FRAME + // It's possible to have an additional thread frame depending on the OS
+                (p.currentJvmVersion() == 8 ? "" : "ThreadJavaMain;") + // Missing in JDK 8
```
Comment (apangin) [inline test/test/stackwalker/StackwalkerTests.java:63]: Checking JVM version is not necessary here. These frames are not guaranteed to present in the particular JDK version anyway. Since this is a regexp, you can just insert a placeholder for an optional frame.

File: test/native/libs/jninativestacks.c (hunk 3 above)
Comment (Baraa-Hasheesh) [reply test/native/libs/jninativestacks.c]: I mean that stack walker here should do complete unwind,
However the name leafFrame is also good, I will use it

File: test/test/stackwalker/StackwalkerTests.java (hunk 6 above)
Comment (Baraa-Hasheesh) [reply test/test/stackwalker/StackwalkerTests.java]: handled here [4432306](https://github.com/async-profiler/async-profiler/pull/1321/commits/4432306d6738f5e5355e4d4e4df57927195221fa)

File: test/test/stackwalker/StackGenerator.java (hunk 5 above)
Comment (Baraa-Hasheesh) [reply test/test/stackwalker/StackGenerator.java]: handled here [4432306](https://github.com/async-profiler/async-profiler/pull/1321/commits/4432306d6738f5e5355e4d4e4df57927195221fa)

File: test/one/profiler/test/TestProcess.java (hunk 4 above)
Comment (Baraa-Hasheesh) [reply test/one/profiler/test/TestProcess.java]: logic was removed as we're no longer checking JVM version [4432306](https://github.com/async-profiler/async-profiler/pull/1321/commits/4432306d6738f5e5355e4d4e4df57927195221fa)

File: test/native/libs/jninativestacks.c (hunk 3 above)
Comment (Baraa-Hasheesh) [reply test/native/libs/jninativestacks.c]: handled here [4432306](https://github.com/async-profiler/async-profiler/pull/1321/commits/4432306d6738f5e5355e4d4e4df57927195221fa)

File: test/native/libs/jninativestacks.c (hunk 2 above)
Comment (Baraa-Hasheesh) [reply test/native/libs/jninativestacks.c]: handled here [4432306](https://github.com/async-profiler/async-profiler/pull/1321/commits/4432306d6738f5e5355e4d4e4df57927195221fa)

File: test/native/libs/jninativestacks.c (hunk 1 above)
Comment (Baraa-Hasheesh) [reply test/native/libs/jninativestacks.c]: handled here [4432306](https://github.com/async-profiler/async-profiler/pull/1321/commits/4432306d6738f5e5355e4d4e4df57927195221fa)
